src/compile/mark/bar.ts

```diff
@@ -17,6 +17,10 @@ export namespace bar {
 
     const stack = model.stack();
     const xFieldDef = model.encoding().x;
+    const x2FieldDef = model.encoding().x2;
+
+    const xIsMeasure = isMeasure(xFieldDef) || isMeasure(x2FieldDef);
```

Comment:
We still need to refactor `bar` and group output by vega properties, but maybe it's better to do this in a follow-up PR.